presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -325,6 +330,16 @@ public TaskStats getTaskStats()
             elapsedTime = new Duration(0, NANOSECONDS);
         }
 
+        synchronized (cumulativeMemoryStatsLock) {
```

Comment:
Instead of doing the integration here, I think you should do it in `DriverContext.startProcessTimer` and then roll it up through the pipeline and task contexts. That will be much more accurate, as there aren't any guarantees about how frequently this method is called.